appcenter/android/src/main/java/com/microsoft/appcenter/reactnative/appcenter/AppCenterReactNativeListener.java

```diff
@@ -27,29 +27,24 @@ public final void setReactApplicationContext(ReactApplicationContext reactApplic
 
     @Override
     public void acquireAuthToken(AuthTokenCallback callback) {
-        ReactNativeUtils.logInfo("Auth token requested from native code.");
-        sendEvent(ON_ACQUIRE_AUTH_TOKEN_EVENT);
+        ReactNativeUtils.logDebug("Auth token requested from native code.");
+        mHasPendingEvent = true;
+        sendEvent(ON_ACQUIRE_AUTH_TOKEN_EVENT, null);
         mCallback = callback;
     }
 
-    private void sendEvent(String eventType) {
-        if (mReactApplicationContext != null) {
-            if (mHasPendingEvent) {
-                mReactApplicationContext
-                        .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
-                        .emit(eventType, null);
-                mAuthTokenEventSent = true;
-            } else {
-                mHasPendingEvent = true;
-            }
+    private void sendEvent(String eventType, Object data) {
+        if (mAuthTokenEventListenerSet && mHasPendingEvent) {
+            mReactApplicationContext
+                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                    .emit(eventType, data);
```

Comment:
add mHasPending = false after emitting.
```suggestion
                    .emit(eventType, data);
        mHasPendingEvent = false;
```